### DISCUSSION TITLE
Emojis not showing up

### DISCUSSION THREAD

**Discussion by Varmod** [AvaloniaUI/Avalonia#12469] [Q&A]
Hi,

I'm having trouble getting the Noto Color Emoji font running in Avalonia. The app runs but the emojis just show as an empty place, text shows up okay.

```xaml
<FontFamily x:Key="NotoColorEmoji">avares://App/Assets/Fonts/#Noto Color Emoji</FontFamily>

<TextBlock Text="🇺🇸 🇩🇪 😀 abc" FontFamily="{StaticResource NotoColorEmoji}" FontSize="20" Background="AliceBlue"/>
```

shows up like this:

![image](https://github.com/AvaloniaUI/Avalonia/assets/52281562/1f8f778f-551d-48e4-a760-4758414c5423)

Any idea what I could be doing wrong?

**Comment by timunie** [comment]
which Avalonia version?

**Reply by Varmod** [reply]
11.0.2

**Reply by timunie** [reply]
okay just installed that font locally and your string doesnt work in MS Word either. So I guess the given char is not there at all. See:

![image](https://github.com/AvaloniaUI/Avalonia/assets/47110241/fc8581b8-e5a4-4d24-b27d-33948f40040e)

**Reply by Varmod** [reply]
Hi, thanks for the observation. It does look like a problem with the font itself

**Reply by Skyedra** [reply]
A bit belated, but I don't think this is a problem with the font itself.  It appears to be a [limitation of colored fonts support in many applications](https://www.colorfonts.wtf/#section4) -- including possibly both word and avalonia.

So for example, using "[Twitter Color Emoji SVGinOT](https://github.com/13rac1/twemoji-color-font/releases)" which has fallback monochrome will render the symbols, just in black & white instead of color.  Noto Color Emoji does not have fallback monochrome symbols as far as I can tell (these are in regular Noto Emoji), which is why some (all?) MS Word versions and Avalonia don't render them.  No monochrome fallback, no render.

What is interesting is that the system fallback emoji does render color emoji just fine in Avalonia (however, depending on user's system -- user may not have such a font installed systemwide, making relying on the fallback less ideal).  I am guessing this uses a different subsystem.

The good news is [Skia did add support for OT-SVG](https://groups.google.com/g/skia-discuss/c/5qiMEM40o50) in 2022ish, but skiasharp stable is on [fairly old skia](https://github.com/mono/SkiaSharp/wiki/Versioning) milestones ([2019ish](https://chromiumdash.appspot.com/schedule)).  Skiasharp 3 will be at least m116, of 2023, so once SkiaSharp 3 is stable and Avalonia uses it, in theory we'll have color emoji support...?

**Reply by Gillibald** [reply]
You can use SkiaSharp V3 with Avalonia today. We have recently added support for it.
